src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/PathMapper.java

```diff
@@ -61,7 +62,14 @@ public org.openkilda.messaging.info.network.Path map(org.openkilda.pce.Path path
             nodes.add(new PathNodePayload(lastSegment.getDestSwitchId(), lastSegment.getDestPort(), null));
         }
 
+        if (path.getProtectedPath() != null && path.getProtectedPath().getProtectedPath() != null) {
```

Comment:
Why do we have a protected path inside the protected path?

Reply:
a protected path is actually the same type as a path. So a Path type contains a Path type variable; there are checks that don't allow this kind of recursion